modules/calib3d/src/usac/essential_solver.cpp

```diff
@@ -239,7 +239,8 @@ class EssentialMinimalSolver5ptsImpl : public EssentialMinimalSolver5pts {
             // (5) Compute the left eigenvectors of the action matrix
             Eigen::EigenSolver<Eigen::Matrix<double, 10, 10>> eigensolver(action_mat_eig);
             const Eigen::VectorXcd &eigenvalues = eigensolver.eigenvalues();
-            const auto * const eig_vecs_ = (double *) eigensolver.eigenvectors().real().data();
+            const Eigen::MatrixXcd eigenvectors = eigensolver.eigenvectors();
+            const auto * const eig_vecs_ = (double *) eigenvectors.data();
```

Comment:
I think dropping the `.real()` call is safe because of the code below (line 277):
```cpp
                    const int eig_i = 20 * i + 12; // eigen stores imaginary values too
```
The code appears to expect to have to skip doubles for the complex values.

Admittedly, I couldn't find a test case that exercised this code path to validate correctness.

Reply:
Call this out in the commit/PR message/description so it's obvious that it is intentional to future sleuths?

Reply:
Good idea. Done.